rotkehlchen/exchanges/ftx.py

```diff
@@ -0,0 +1,477 @@
+                    amount=amount,
+                    usd_value=amount * usd_price,
+                )
+            except UnknownAsset as e:
+                self.msg_aggregator.add_warning(
+                    f'Found FTX balance result with unknown asset '
+                    f'{e.asset_name}. Ignoring it.',
+                )
+                continue
+            except UnsupportedAsset as e:
+                self.msg_aggregator.add_warning(
+                    f'Found FTX balance result with unsupported asset '
+                    f'{e.asset_name}. Ignoring it.',
+                )
+                continue
+            except (DeserializationError, KeyError) as e:
+                msg = str(e)
+                if isinstance(e, KeyError):
+                    msg = f'Missing key entry for {msg}.'
+                self.msg_aggregator.add_error(
+                    'Error processing a FTX account balance. Check logs '
+                    'for details. Ignoring it.',
+                )
+                log.error(
+                    'Error processing a FTX balance',
+                    error=msg,
+                )
+                continue
+
+        return dict(returned_balances), ''
+
+    def query_online_trade_history(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> List[Trade]:
+
+        raw_data = self._api_query('fills', start_time=start_ts, end_time=end_ts)
+        log.debug('FTX buys/sells history result', results_num=len(raw_data))
+
+        trades = []
+        for raw_trade in raw_data:
+            try:
+                trade = trade_from_ftx(raw_trade)
+            except UnknownAsset as e:
+                self.msg_aggregator.add_warning(
+                    f'Found FTX transaction with unknown asset '
+                    f'{e.asset_name}. Ignoring it.',
+                )
+                continue
+            except UnsupportedAsset as e:
+                self.msg_aggregator.add_warning(
+                    f'Found FTX trade with unsupported asset '
+                    f'{e.asset_name}. Ignoring it.',
+                )
+                continue
+            except (DeserializationError, KeyError) as e:
+                msg = str(e)
+                if isinstance(e, KeyError):
+                    msg = f'Missing key entry for {msg}.'
+                self.msg_aggregator.add_error(
+                    'Error processing a FTX trade. Check logs '
+                    'for details. Ignoring it.',
+                )
+                log.error(
+                    'Error processing a FTX trade',
+                    trade=raw_trade,
+                    error=msg,
+                )
+                continue
+
+            # limit coinbase trades in the requested time range here since there
+            # is no argument in the API call
+            if trade and trade.timestamp >= start_ts and trade.timestamp <= end_ts:
+                trades.append(trade)
```

Comment:
Is there really no argument in the api? The entire pagination functionality is based on timestamps.

Also /s/coinbase/ftx

Reply:
This is an error because I copied the skeleton from the Coinbase API and starting making changes to the needed parts, but sometimes I followed the complex calls and forgot to go back. From this PR I've learnt to copy less and take more care to this details